tests/unit/lib/Gocdb_Services/ServiceServiceTest.php

```diff
@@ -0,0 +1,195 @@
+    $tables = simplexml_load_file($fixture);
+
+    foreach($tables as $tableName) {
+      $sql = "SELECT * FROM ".$tableName->getName();
+      $result = $con->createQueryTable('results_table', $sql);
+      if($result->getRowCount() != 0){
+        throw new RuntimeException("Invalid fixture. Table has rows: ".$tableName->getName());
+      }
+    }
+  }
+
+  public function createValidationEntities(){
+    
+    $site = TestUtil::createSampleSite("TestSite");
+    $this->em->persist($site);
+
+    $service = TestUtil::createSampleService("TestService1");
+    $this->em->persist($service);
+    $service->setParentSiteDoJoin($site);
+
+    $user = TestUtil::createSampleUser("Test", "Testing", "/c=test");
+    $this->em->persist($user);
+    $user->setAdmin(TRUE);
+
+    $roleActionMappingService = new org\gocdb\services\RoleActionMappingService();
+    $roleActionAuthService = new org\gocdb\services\RoleActionAuthorisationService($roleActionMappingService);
+    $roleActionAuthService->setEntityManager($this->em);
+
+    $this->em->flush();
+
+    $serviceService = new org\gocdb\services\ServiceService();
+    $serviceService->setEntityManager($this->em);
+    $serviceService->setRoleActionAuthorisationService($roleActionAuthService);
+
+    return array($service, $user, $serviceService);
+  }
+  /**
+   * Check the basics -
+   * Duplicates some simple testing from ExtensionsTest
+   */
+  public function testValidateProperty(){
+    print __METHOD__ . "\n";
+
+    list ($service, $user, $serviceService) = $this->createValidationEntities();
+   
+    $values[0] = array("ValidName1","ValidValue");
+    $values[1] = array("ValidName2","<ValidValue><ValidValue>");
+
+    $this->assertTrue($serviceService->addProperties($service, $user, $values) == NULL);
+
+    $this->assertTrue(count($properties = $service->getServiceProperties()) == 2);
+
+    $this->assertTrue(
+      $serviceService->deleteServiceProperties($service, $user, $properties->toArray())
+                      == NULL);
+
+    $this->assertTrue(count($properties = $service->getServiceProperties()) == 0);
+  }
+  /**
+   * Check that validation of property name is operating as expected
+   * Added to test code changes to pass through "<>" chars
+   * @depends testValidateProperty
+   */
+  public function testValidatePropertyNameFails(){
+    print __METHOD__ . "\n";
+
+    list ($service, $user, $serviceService) = $this->createValidationEntities();
+   
+    $values[0] = array("<Invalid>","<valid>");
+
+    $this->setExpectedException('Exception');
+
+    $serviceService->addProperties($service, $user, $values);
+    
+  }
+  /**
+   * Check that validation of property value is operating as expected
+   * @depends testValidateProperty
+   */
+  public function testValidatePropertyValueFails(){
+    print __METHOD__ . "\n";
+
+    list ($service, $user, $serviceService) = $this->createValidationEntities();
+   
+    $values[0] = array("Valid","'Not Valid'");
```

Comment:
Is this array of the form ("Property Name", Property Value")? Its worth a comment as to why this case is invalid.

Reply:
Added comment